utils/queuer.go

```diff
@@ -67,13 +79,19 @@ func (q GeneralKVQueue) Dequeue(value codec.ProtoMarshaler, filter ...func(value
 		return false
 	}
 
-	if len(filter) > 0 && !filter[0](value) {
-		return false
-	}
+	pass, stop := filter(value)
+	switch {
+	case pass:
+		q.store.Delete(KeyFromBz(iter.Key()))
 
-	q.store.Delete(KeyFromBz(iter.Key()))
+		return true
+	case stop:
+		return false
```

Comment:
The case when both are true is invalid. We could consider using an enum instead of two bools. But otherwise, maybe check if `stop` is `true` first so we're on the safer side?

Reply:
interesting. I like that. Let me give it a try.